src/middleware/socket/handler.js

```diff
@@ -28,12 +15,25 @@ export default function setupSocketHandler(app, options, {
   }
 
   return function(socket) {
-    const authenticate = (data, callback = () => {}) => {
+    const authenticate = function (data, callback = () => {}) {
       service.create(data, { provider })
-        .then( ({ token }) => {
+        .then(jwt => app.authentication.authenticate(jwt))
+        .then(result => {
+          if(!result.authenticated) {
+            throw new errors.NotAuthenticated('Authentication was not successful');
+          }
+
+          const { token } = result;
+          const connection = feathersParams(socket);
+
           debug(`Successfully authenticated socket with token`, token);
 
-          feathersParams(socket).token = token;
+          // Add the token to the connection so that it shows up as `params.token`
+          connection.token = token;
+
+          app.emit('login', result, {
```

Comment:
@daffl sounds good. Hopefully, because it's not on a service people won't be confused about it being dispatched outside of the server. I agree with you, I don't think they should be available to a client.